### DISCUSSION TITLE
GlobalStorken

### DISCUSSION THREAD

**Discussion by cetfu** [storkenjs/storken#5] [Q&A] [answered]
Merhaba, GlobalStorken kullanımına dair herhangi bir örnek göremedim. Kaynak koduna bakmama rağmen herhangi bir şey anlamadım. Bir örnek verebilir misiniz?

**Comment by cetfu** [comment]
Örnek yapılan projeyi biraz kurcalayınca nasıl yapmam gerektiğini az da olsa anladım. Üzerine örnek yapmaya çalışınca nasıl kullanmam gerektiğini çözdüğümü düşünüyorum. Paketi inşallah daha çok kişi görür ve kullanır 🍀

**Comment by K-NRS** [comment, marked as answer]
Merhaba,

`GlobalStorken` aslında oluşturulan Store için her türlü işleme her yerden "Global olarak" erişim sağlamaya yaran ve aynı zamanda aslında konfigürasyonlarıyla birlikte Storken ile oluşturulan bütün stateleri de içinde bulunduran nesnedir. Projeniz belli bir aşamaya gelmediği sürece pek kullanma ihtiyacı duyulmaz.

Yakında kullanımıyla ilgili daha açıklayıcı belgeler repoda yer alacak. İlginiz için teşekkürler.

**Reply by cetfu** [reply]
Hocam aslında ben realtime chat uygulaması yapmak istiyordum(mesaj silme, mesaj güncelleme gibi fonksiyonlar da bulunacak) ancak mesajları stateye kaydedip maplediğimde daha sonra mesajın silindiğinde mesajın ekrandan gitmesi için re-render yapmak gerekmiş. Ancak ben useEffect içinde socket servere istek atıp mesajları çekiyorum. Mesajları da stateye yazdığım için state sürekli set edilerek(rerender olması için useEffectin dependency kısmına stateyi veriyorum ) clientte performans sorunu oluşuyor. Bu durumu storken kullanarak aşabilir miyim? Eğer aşabilirsem nasıl yapmam gerekecek? Sorunun içinden çıkamadığım için buraya yazdım kusura bakmayın

**Reply by K-NRS** [reply]
Websocket mesajlarını useEffect ve componentten bağımsız bir yerde dinlerken `GlobalStorken` ile setleyebilirsin.
```js
ws.onMessage(function(deletedMessage) {
  GlobalStorken.set('messages', messages => 
    messages.filter(message => message.id !== deletedMessage.id)
  )
})
```

gibi bir kullanımla mesajları tek seferde güncelleyebilirsin. WebSocket üzerinden eğer bütün mesajlar geliyor yine `GlobalStorken.set('messages', messages)` şeklinde tek seferde bütün mesajları güncelleyebilirsin.

Bonus bilgi: Storken ayrıca getter & setter özellikleri ve pluginleriyle useEffect yönetmeyi önemli ölçüde azaltır veya bazı durumlarda tamamen kaldırır. 🙂

**Reply by cetfu** [reply]
Çok teşekkür ederim 🙏
